### DISCUSSION TITLE
Generalize the scale gizmo handles setting in the touch screen interface settings

### DISCUSSION THREAD

**Discussion by LunaCapra** [godotengine/godot-proposals#8498] [Editor]
As it stands, most editor gizmo billboards/handles are fairly small, making it hard for me to use them.

I think it would be beneficial to have a multiplier for the scale of billboard gizmos.
It could potentially be a slider, similar to what [Unity has](https://docs.unity3d.com/Manual/GizmosMenu.html) for when "3D Gizmos" is enabled.

**Comment by Calinou** [comment]
We probably also want to have distance fade properties for gizmo icons, so that distant gizmos can be hidden to be less obtrusive (since their size is fixed).

Alternatively, we can make gizmo icon size depend on distance, but this will require a custom shader so that the size is kept to a reasonable value at its minimum size.

**Reply by LunaCapra** [reply]
I'm not sure how exactly the current gizmos are set up but this would probably also require sub-gizmos or additional handles positioned alongside the primary handles to take the scale into account as well.

The main example I could think of are the handles for a Path3D point.

**Reply by Calinou** [reply]
> I'm not sure how exactly the current gizmos are set up but this would probably also require sub-gizmos or additional handles positioned alongside the primary handles to take the scale into account as well.

When drawing, each gizmo retrieves a material ([example](https://github.com/godotengine/godot/blob/fa4a65387ea41506ecc519fc61a06da110dce51c/editor/plugins/path_3d_editor_plugin.cpp#L275-L279)) generated in the constructor ([example](https://github.com/godotengine/godot/blob/fa4a65387ea41506ecc519fc61a06da110dce51c/editor/plugins/path_3d_editor_plugin.cpp#L820-L822)).

Here, [`EditorNode3DGizmoPlugin::create_handle_material()`](https://github.com/godotengine/godot/blob/fa4a65387ea41506ecc519fc61a06da110dce51c/editor/plugins/node_3d_editor_gizmos.cpp#L941-L962) would have to be modified.

**Reply by LunaCapra** [reply]
I've been wondering, if there were more settings like you suggested - e.g. distance fade - where would these options go?
Currently there is `View > Gizmos` in the 3D editor to toggle individual Node gizmos.
But there is also `Perspective > View Gizmos` to turn them all of entirely.
There's also the entirely separate popup window under `View > Settings`, which seems to control the view camera rather than general view settings.

**Comment by LunaCapra** [comment]
I would like to note that it's been brought to my attention that there is a setting seemingly specific to the Android editor that *does* scale the gizmo handle size already.

`Interface > Touchscreen > Scale Gizmo Handles`

However, I do feel it would still be preferable to make this a general setting and expand upon it with the suggested distance fade and maybe other features.
